packages/tizen_notification/lib/notification.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:flutter/services.dart';
+import 'package:tizen_app_control/app_control.dart';
+
+import 'src/types.dart';
+
+export 'package:tizen_app_control/app_control.dart';
+
+export 'src/enums.dart';
+export 'src/types.dart';
+
+// ignore: public_member_api_docs
+class TizenNotificationDetails {
+  /// [TizenNotificationDetails] contains detail information about the notification
+  /// being sent.
+  ///
+  /// [images] only supports type [List<NotificationImage>] and [NotificationImage].
+  const TizenNotificationDetails({
+    this.images,
+    this.properties,
+    this.appControl,
+    this.vibration,
+    this.sound,
+    this.displayApplist = DisplayApplist.all,
+    this.onGoing = false,
+  });
+
+  /// A single or a list of [NotificationImage].
+  ///
+  /// [NotificationImage] refers to any image shown on the notification panel.
+  final dynamic images;
+
+  /// Properties that configure how the system handles notification at
+  /// certain scenarios.
+  ///
+  /// [properties] can have multiple [Property] values with the bitwise OR operator.
+  /// For example, [Property.disableAppLaunch] | [Property.volatileDisplay].
+  final int? properties;
+
+  /// A set of information used by app control to launch other applications.
+  final AppControl? appControl;
+
+  /// The vibration triggered when a notification is shown on the panel.
+  final NotificationVibration? vibration;
+
+  /// The sound played when a notification is shown on the panel.
+  final NotificationSound? sound;
+
+  /// The destination app that displays notification.
+  ///
+  /// The notification can be sent to multiple apps by listing values with the
+  /// bitwise OR operator.
+  /// For example, [DisplayApplist.tray] | [DisplayApplist.indicator].
+  final int? displayApplist;
+
+  /// If [onGoing] is true, the user is not able to delete the notification.
+  ///
+  /// Currently, only the common profile supports this operation.
+  final bool onGoing;
+
+  /// Returns [TizenNotificationDetails] memeber fields in a map format.
+  Map<String, dynamic> toMap() => <String, dynamic>{
+        'displayApplist': displayApplist,
+        'properties': properties,
+        'vibration': vibration?.toMap(),
+        'sound': sound?.toMap(),
+        'images': _getImages(images)
+            ?.map((NotificationImage image) => image.toMap())
+            .toList(),
```

Comment:
This is not null safe.

Reply:
I can't find which part, could you explain in detail?